test/spoof/spoof.go

```diff
@@ -241,6 +242,16 @@ func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker) (*Res
 				sc.logf("Retrying %s for TCP timeout %v", req.URL.String(), err)
 				return false, nil
 			}
+
+			// Repeat the poll on `connection refused` errors, which are usually transient Istio errors.
+			// The alternative for the string check is:
+			// 	errNo := (((err.(*url.Error)).Err.(*net.OpError)).Err.(*os.SyscallError).Err).(syscall.Errno)
+			// if errNo == syscall.Errno(0x6f) {...}
+			// But with assertions, of course.
+			if strings.Contains(err.Error(), "connect: connection refused") {
+				sc.logf("Retrying %s for connection refused %v", req.URL.String(), err)
+				return false, nil
+			}
```

Comment:
Can we by any chance catch this via

```golang
if err, ok := err.(net.Error); ok && err.Temporary() {}
```

Or does this not fall into that category?

Reply:
Unfortunately connection refused is not temporary or timeout :crying_cat_face: